pkg/gadgets/dns/gadget.go

```diff
@@ -154,61 +162,48 @@ func (t *Trace) Start(trace *gadgetv1alpha1.Trace) {
 		}
 	}
 
-	genKey := func(namespace, podname string, pid uint32) (key string, err error) {
-		key = namespace + "/" + podname
-
-		var netns1, netns2 uint64
-		netns1, err = containerutils.GetNetNs(int(pid))
-		if err == nil {
-			netns2, err = containerutils.GetNetNs(os.Getpid())
-		}
-		if err == nil {
-			if netns1 == netns2 {
-				key = "host"
-			}
+	genKey := func(container *pb.ContainerDefinition) string {
+		if container.Netns == t.netnsHost {
+			return "host"
 		}
-		log.Infof("DNS gadget: generate key %q from pod (%q %q pid:%d) (netns:%v host-netns:%v)", key, namespace, podname, pid, netns1, netns2)
-		return key, err
+		return container.Namespace + "/" + container.Podname
 	}
 
-	attachContainerFunc := func(namespace, podname string, pid uint32) error {
-		key, err := genKey(namespace, podname, pid)
+	attachContainerFunc := func(container *pb.ContainerDefinition) error {
```

Comment:
I know it's not strictly related to this PR, but why is it returning an error if it's always returning `nil`?